### PR TITLE
Fix PasswordCredential(form) constructor for password change forms

### LINKED ISSUE #34: PasswordCredential(HTMLFormElement form) may fail in valid password change form cases
The following form is a valid change password form that fails to be parsed by the PasswordCredential(HTMLFormElement form) constructor
```
<form action="https://example.com/login" method="POST" id="theForm">
  <label for="username">Username</label>
  <input type="text" id="username" name="username" autocomplete="username">
  <label for="new-password">New Password</label>
  <input type="text" id="new-password" name="new-password" autocomplete="new-password">
  <label for="old-password">Old Password</label>
  <input type="text" id="old-password" name="old-password" autocomplete="current-password">
  <input type="submit">
</form>
```

The PasswordCredential(HTMLFormElement form) constructor is sensitive to the order of "current-password" and "new-password" fields. It takes the last one and stores it in the credential. In the example above, the old password gets saved, not the new one.

### PR DESCRIPTION
**Pull request by battre**
Change the PasswordCredential(form) constructor so that for common password-change forms that ask for the user's current-password and new-password only the new-password is adopted.

Fixes #34

### CODE REVIEW

File: index.src.html [hunk 1]
```diff
@@ -814,12 +816,21 @@ <h5 id="passwordcredential-form-constructor" algorithm>
...
+                      |name|, and |newPasswordObserved| to true.
+                  :   "<a attr-value>`current-password`</a>"
+                  ::  If |newPasswordObserved| is false,
+                      set |data|'s {{PasswordCredentialData/password}} member's
+                      value to the result of executing |formData|'s
                       {{FormData/get()}} method on |name|, and |passwordName| to
                       |name|.
+
+                      Note: By checking that |newPasswordObserved| is false,
+                      new-password fields take precedence over current-password
```
Comment (mikewest) [inline index.src.html]: Tiny nit: Either ``` `new-password` ``` (as code, with backticks) or "new password" (as prose, without backticks or hyphens).

File: index.src.html [hunk 2]
```diff
@@ -814,12 +816,21 @@ <h5 id="passwordcredential-form-constructor" algorithm>
               1.  If |token| is an <a>ASCII case-insensitive</a> match for one
                   of the following strings, run the associated steps:
 
-                  :   "<a attr-value>`current-password`</a>"
                   :   "<a attr-value>`new-password`</a>"
                   ::  Set |data|'s {{PasswordCredentialData/password}} member's
                       value to the result of executing |formData|'s
+                      {{FormData/get()}} method on |name|, |passwordName| to
+                      |name|, and |newPasswordObserved| to true.
```
Comment (mikewest) [inline index.src.html]: Tiny nit: ``` `true` ```

File: index.src.html [hunk 3]
```diff
@@ -814,12 +816,21 @@ <h5 id="passwordcredential-form-constructor" algorithm>
...
+                  :   "<a attr-value>`current-password`</a>"
+                  ::  If |newPasswordObserved| is false,
```
Comment (mikewest) [inline index.src.html]: Tiny nit: ``` `false` ```

File: index.src.html [hunk 4]
```diff
@@ -814,12 +816,21 @@ <h5 id="passwordcredential-form-constructor" algorithm>
...
+                      set |data|'s {{PasswordCredentialData/password}} member's
+                      value to the result of executing |formData|'s
                       {{FormData/get()}} method on |name|, and |passwordName| to
                       |name|.
+
+                      Note: By checking that |newPasswordObserved| is false,
```
Comment (mikewest) [inline index.src.html]: Tiny nit: ``` `false` ```

File: index.src.html (hunk 1 above)
Comment (battre) [inline index.src.html]: Done.

File: index.src.html (hunk 4 above)
Comment (battre) [inline index.src.html]: Done.

File: index.src.html (hunk 3 above)
Comment (battre) [inline index.src.html]: Done.

File: index.src.html (hunk 2 above)
Comment (battre) [inline index.src.html]: Done.

**Comment by mikewest** [comment]
Still LGTM.

**Pull request merged**